llm_bioinformatics_research/src/UserProfile/UserProfileMenu.js

```diff
@@ -0,0 +1,171 @@
+    localStorage.removeItem('userId');
+    navigate('/login');
+  };
+  const handleThemeChange = async (e) => {
+    const newTheme = e.target.value;
+    setUser((prev) => ({ ...prev, theme: newTheme })); 
+    toggleTheme(newTheme); 
+    await updateUserSettings(newTheme, user.language); 
+  };
+
+  const handleLanguageChange = async (e) => {
+    const newLanguage = e.target.value;
+    setUser((prev) => ({ ...prev, language: newLanguage }));
+  };
+
+  const updateUserSettings = async (theme, language) => {
+    const userData = JSON.parse(localStorage.getItem('userData'));
+    const userId = userData ? userData.id : null;
+  
+    if (!userId) {
+      alert('User ID not found. Please log in again.');
+      return;
+    }
+  
+    try {
+      const response = await fetch('http://localhost:' + port + '/profile', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          userId: userId,
+          profile_photo: userData.profile_photo || '',
+          phone_number: userData.phone_number || '',
+          location: userData.location || '',
+          theme: theme,
+          language: language
+        }),
+      });
+  
+      const data = await response.json();
+      if (!response.ok) {
+        throw new Error(data.message || 'Failed to update settings');
+      }
+  
+      const updatedUserData = { 
+        ...userData, 
+        theme, 
+        language 
+      };
+  
+      localStorage.setItem('userData', JSON.stringify(updatedUserData));
+  
+      console.log('Settings updated successfully:', updatedUserData);
+      alert('Updated successfully!');
+    } catch (error) {
+      console.error('Error updating settings:', error);
+      alert('An error occurred while updating settings');
+    }
+  };
+
+  return (
+    <>
+      <IconButton onClick={handleOpenMenu} aria-label="user profile menu">
+        {user.image ? (
+          <Avatar src={user.image} alt="User Profile" className="user-avatar" />
+        ) : (
+          <PersonOutlineIcon />
+        )}
+      </IconButton>
+      <Menu anchorEl={anchorEl} open={Boolean(anchorEl)} onClose={handleCloseMenu} className='menu'>
+        <Box className="menu-header">
+          <Avatar alt="User Image" src={user.image} className="user-avatar" />
+          <Box sx={{ marginLeft: 2 }}>
+            <Typography variant="body1">{user.name || 'Your name'}</Typography>
+            <Typography variant="body2" color="textSecondary">{'Your name'}</Typography>
+          </Box>
+          <IconButton onClick={handleCloseMenu} size="small" aria-label="close menu">
+            <CloseIcon />
+          </IconButton>
+        </Box>
+        <MenuItem className="menu-item" onClick={() => navigate('/profile')}>
+          <PersonOutlineIcon className="menu-item-icon" /> My Profile
+          <ExpandMoreIcon style={{ marginLeft: 'auto', transform: 'rotate(-90deg)' }} />
```

Comment:
I am not entirely sure if we need the "My Profile" component that is at the corner of this page. It seems like a non-functional button that can be removed. 

![image](https://github.com/user-attachments/assets/e0dcf0b3-3d17-4d05-9d2a-b44c52c268bf)